### ISSUE TITLE
readinessGate doesn't work on 1.1.6 and 1.1.7

### ISSUE THREAD

**Issue by DiegoLelis** [kubernetes-sigs/aws-load-balancer-controller#1241]
I'm using EKS 1.15.
alb: 1.1.7
Following the example: 
```
        kubectl get pod  -o wide
NAME                                READY   STATUS    RESTARTS   AGE     IP             NODE                                          NOMINATED NODE   READINESS GATES
nginx-deployment-587f7ff476-bnvq9   1/1     Running   0          8m48s   10.10.107.13   ip-10-10-100-244.us-west-2.compute.internal   <none>           0/1
nginx-deployment-587f7ff476-vgnkx   1/1     Running   0          8m48s   10.10.82.91    ip-10-10-74-132.us-west-2.compute.internal    <none>           0/1
nginx-deployment-5d5d44bdb9-q9ttt   1/1     Running   0          7m34s   10.10.64.95    ip-10-10-78-85.us-west-2.compute.internal     <none>           0/1
```
https://kubernetes-sigs.github.io/aws-alb-ingress-controller/guide/ingress/pod-conditions/
Tried both alb versions.
I can access the app, but the readinessGate never passes.

**Comment by astrived** [comment]
@DiegoLelis can you add more details on your test top and when it is not working ? @devkid had resolved the known issues in 1.1.7

**Comment by DiegoLelis** [comment]
Hello @astrived 
I'm using the version 1.1.7 with EKS 1.15

This is the error that I see on my pods:
```
  - lastProbeTime: null
    lastTransitionTime: "2020-05-03T11:49:19Z"
    message: corresponding condition of pod readiness gate "target-health.alb.ingress.k8s.aws/nginx-ingress_nginx-service_80"
      does not exist.
    reason: ReadinessGatesNotReady
    status: "False"
    type: Ready
```
I've also attached all the yaml files that I am using it, and the one from the pod with the status, and my alb ingress controller deployment.

[test-readiness-gate.zip](https://github.com/kubernetes-sigs/aws-alb-ingress-controller/files/4570217/test-readiness-gate.zip)

**Comment by shaoxt** [comment]
I saw same issue on version 1.1.6.
The pod is pending for ever.  It won't retry.
Not sure whether 1.1.7 fixed this.

**Comment by sjammy6** [comment]
@DiegoLelis @shaoxt I am also running into the same issue, did you find anything? 
I am running EKS 1.14 and ALB 1.1.7

**Comment by shaoxt** [comment]
> @DiegoLelis @shaoxt I am also running into the same issue, did you find anything?
> I am running EKS 1.14 and ALB 1.1.7

Can you try this in EKS 1.15 + ALB 1.1.7 ?

**Comment by sjammy6** [comment]
@shaoxt I can definitely try that.
 I just want to know, is it working for you with EKS 1.15 + ALB 1.1.7?

**Comment by shaoxt** [comment]
Yeah

**Comment by sjammy6** [comment]
@shaoxt Thank you. I just can not find any proof that why it is not working with EKS 1.14. 
@devkid any guidance on why readinessGates not working with EKS1.14 + ALB 1.1.7 will be really helpful.

**Comment by jmriebold** [comment]
Just discovered an important wrinkle to setting up readinessGates that isn't featured in the documentation: if you use named ports, e.g. `http` in the following `Service`:

```yaml
kind: Service
apiVersion: v1
metadata:
  name: ingress-test
  namespace: test
  labels:
    app: ingress-test
spec:
  ports:
    - name: http
      protocol: TCP
      port: 80
      targetPort: traffic
  selector:
    app: ingress-test
```

Then your readinessGate must reference the port _name_, not the port number itself, e.g. `conditionType: target-health.alb.ingress.k8s.aws/ingress-test_ingress-test_http`, not `conditionType: target-health.alb.ingress.k8s.aws/ingress-test_ingress-test_80`.

Once I changed that, it worked perfectly.

**Comment by sjammy6** [comment]
@jmriebold Thank you for letting us know. Now it all makes sense, it started for me after I changed `conditionType: target-health.alb.ingress.k8s.aws/load-balancer-any-tg-ready` that is supported by `v1.1.8`.

**Comment by M00nF1sh** [comment]
Hi guys,
we have released a complete new version v2.0.0, the readinessGate implementation have been changed as well.
Also, you no longer needs to manually add these readinessGates any more, we have a webhook to do the job.
please check out https://kubernetes-sigs.github.io/aws-load-balancer-controller/guide/controller/pod_readiness_gate/.

closing this, feel free to reopen or create new issue if you encounter any readinessGate issue in v2.0.0
